lib/screen/chat/chat_thread_page.dart

```diff
@@ -167,6 +171,18 @@ class _ChatThreadPageState extends State<ChatThreadPage> {
     final digest = hmacSha256.convert(utf8.encode(canonicalString));
     final sig = bytesToHex(digest.bytes);
     header["X-Api-Signature"] = sig;
+
+    final pubKey =
+        await _payload.wallet.getTezosPublicKey(index: _payload.index);
+    final authSig = await injector<TezosService>().signMessage(_payload.wallet,
+        _payload.index, Uint8List.fromList(utf8.encode(timestamp.toString())));
+    final token = await injector<ChatApi>().getToken({
```

Comment:
we should cache the token, only request to get new token when the old jwt is expired. Refer to `auth_service` for more information.